### PR TITLE
docs(autonomi): add example Rust code

### PR DESCRIPTION
**Pull request by b-zee**

### CODE REVIEW

File: autonomi/examples/put_and_dir_upload.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Default wallet of testnet.
+    let key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+    let client = Client::connect(&["/ip4/127.0.0.1/udp/1234/quic-v1".parse()?]).await?;
+    let wallet = Wallet::new_from_private_key(EvmNetwork::ArbitrumSepolia, key)?;
+
+    // Put and fetch data.
+    let data_addr = client
+        .data_put(Bytes::from("Hello, World"), wallet.clone().into())
```
Comment (mickvandijke) [inline autonomi/examples/put_and_dir_upload.rs]:
```suggestion
        .data_put(Bytes::from("Hello, World"), &wallet.into())
```

File: autonomi/README.md [hunk 2]
```diff
@@ -9,9 +9,38 @@ Connect to and build on the Autonomi network.
...
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Default wallet of testnet.
+    let key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+    let client = Client::connect(&["/ip4/127.0.0.1/udp/1234/quic-v1".parse()?]).await?;
+    let wallet = Wallet::new_from_private_key(EvmNetwork::ArbitrumSepolia, key)?;
+
+    // Put and fetch data.
+    let data_addr = client
+        .data_put(Bytes::from("Hello, World"), wallet.clone().into())
```
Comment (mickvandijke) [inline autonomi/README.md]:
```suggestion
        .data_put(Bytes::from("Hello, World"), &wallet.into())
```

File: autonomi/README.md [hunk 3]
```diff
@@ -9,9 +9,38 @@ Connect to and build on the Autonomi network.
...
+```rust
+use autonomi::{Bytes, Client, EvmNetwork, Wallet};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Default wallet of testnet.
+    let key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+    let client = Client::connect(&["/ip4/127.0.0.1/udp/1234/quic-v1".parse()?]).await?;
+    let wallet = Wallet::new_from_private_key(EvmNetwork::ArbitrumSepolia, key)?;
```
Comment (mickvandijke) [inline autonomi/README.md]:
```suggestion
    let wallet = Wallet::new_from_private_key(Default::default(), key)?;
```

Maybe we should use the default `EvmNetwork` here and add a separate section for setting the wallet up with a different/custom network?

File: autonomi/README.md (hunk 2 above)
Comment (mickvandijke) [reply autonomi/README.md]: Ah this is actually the same thing under the hood, so doesn't really matter!

File: autonomi/examples/put_and_dir_upload.rs (hunk 1 above)
Comment (b-zee) [reply autonomi/examples/put_and_dir_upload.rs]: This yields

```
mismatched types
  expected enum `autonomi::client::payment::PaymentOption`
found reference `&_`
```

Comment (mickvandijke) [reply autonomi/examples/put_and_dir_upload.rs]: Oh sorry!

Should be:

```rust
        .data_put(Bytes::from("Hello, World"), (&wallet).into())
```

But like I said in my other comment, this is the same as cloning under the hood in the `From` implementation, so I suppose it doesn't matter too much.

**Review by mickvandijke — APPROVED** [review]
Nice work!

**Pull request merged**